controllers/release/release_adapter.go

```diff
@@ -228,24 +229,39 @@ func (a *Adapter) EnsureSnapshotEnvironmentBindingIsCreated() (results.Operation
 // EnsureTargetContextIsSet is an operation that will ensure that the targetContext of the adapter is not nil. It will
 // set it to a context from the workspace in the ReleasePlan if one is defined or to its own context if not.
 func (a *Adapter) EnsureTargetContextIsSet() (results.OperationResult, error) {
-	if a.targetContext == nil {
+	// If the target namespace/workspace is not set at the release status we should get the target from the ReleasePlan
+	if a.release.Status.Target == (kcp.NamespaceReference{}) {
+		patch := client.MergeFrom(a.release.DeepCopy())
 		releasePlan, err := a.getReleasePlan()
 		if err != nil {
-			patch := client.MergeFrom(a.release.DeepCopy())
 			a.release.MarkInvalid(v1alpha1.ReleaseReasonReleasePlanValidationError, err.Error())
 			return results.RequeueOnErrorOrStop(a.client.Status().Patch(a.context, a.release, patch))
 		}
 
-		if releasePlan.Spec.Target.Workspace == "" {
-			a.logger.Info("Running in a traditional kubernetes environment")
-			a.targetContext = a.context
-		} else {
-			a.logger.Info("Running in a KCP environment")
-			a.targetContext = logicalcluster.WithCluster(a.context, logicalcluster.New(releasePlan.Spec.Target.Workspace))
+		a.release.Status.Target = releasePlan.Spec.Target
+		err = a.client.Status().Patch(a.context, a.release, patch)
+		if err != nil {
+			return results.RequeueWithError(err)
 		}
 
+	}
+
+	if a.release.Status.Target.Workspace != "" {
+		// If this is a non-KCP cluster and the target workspace is set we should stop the release
+		// otherwise it fails at the release_controller because it can not switch to a non existent
+		// context
+		if _, ok := logicalcluster.ClusterFromContext(a.context); !ok {
+			patch := client.MergeFrom(a.release.DeepCopy())
+			a.release.MarkInvalid(v1alpha1.ReleaseReasonReleasePlanValidationError, fmt.Sprintf("KCP Workspace %s does not exist", a.release.Status.Target.Workspace))
```

Comment:
The error should probably be something like `ReleasePlan Workspace set on a non-KCP environment`.

Reply:
But it also fails if the workspace does not exist - I.E. Is a KCP cluster but the name has a typo. What about:
_ReleasePlan Workspace inexistent or set on a non-KCP environment_